### PR TITLE
[plugin.video.mediathekview] 1.0.2+matrix.1

### PR DESCRIPTION
**Pull request by codingPF**
### Description
Update for the plugin.video.mediathekview plugin

Main Topics
- [fix] #125 fix for add to favorits
- [new] #56 simple black list
- [other] New UI icons and adjusted view (Wall view)
- [other] New extended search to filter all criteria
- [other] Database structure between mysql and sqlite aligned
- [other] Navigation for all channels moved into channel view
- [other] Settings split into general (UI settings) and advanced settings for UI behaviour
- [other] Code restructring

### Checklist:

- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practice but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-plugins/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.
- If you see no activity on your PR after a week (so at least one weekend has passed) then please go to the #kodi-dev freenode IRC channel to reach out to the team

### CODE REVIEW

File: plugin.video.mediathekview/resources/lib/mvupdate.py [hunk 1]
```diff
@@ -13,405 +13,45 @@
...
-        """
-        Returns `True`if either this instance is not the registered
-        instance or Kodi is shutting down
-        """
-        return False
-
-
-class UpdateApp(AppLogger):
+class UpdateApp():
     """ The standalone updater application class """
```
Comment (enen92) [inline plugin.video.mediathekview/resources/lib/mvupdate.py:27]: Sorry You'll have to drop this. Add-ons are not allowed to parse the add-on.xml nor to control any update mechanisms aside from kodi

File: plugin.video.mediathekview/resources/lib/updateFileDownload.py [hunk 2]
```diff
@@ -0,0 +1,247 @@
+# -*- coding: utf-8 -*-
```
Comment (enen92) [inline plugin.video.mediathekview/resources/lib/updateFileDownload.py:1]: This functionality might be a bit gray I'd say. @Lunatixz can you double check?

**Review by enen92 — CHANGES_REQUESTED** [review]
At least the update functionality has to go. Thanks for the update

File: plugin.video.mediathekview/resources/lib/updateFileDownload.py (hunk 2 above)
Comment (Lunatixz) [reply plugin.video.mediathekview/resources/lib/updateFileDownload.py:1]: @enen92 Yea, IMO a little more than gray. This type of behavior is not permitted in Kodi's repository for a few reasons. Mainly a database file is binary which is not permitted. Self-updating a file via download is also not permitted; I assume the database contains a list of online media? This is also not permitted unless through an API or a local (non-binary) file included with the review. If hosted the API would be vetted by the team for legitimate ownership of any copyrighted material.

File: plugin.video.mediathekview/resources/lib/kodi/kodiaddon.py [hunk 3]
```diff
@@ -96,18 +97,59 @@ def run_builtin(self, builtin):
...
+        return self.language(msgid);
+
+    def getSkinName(self):
+        return xbmc.getSkinDir();
+
+    def getCurrentViewId(self):
+        window = xbmcgui.Window(xbmcgui.getCurrentWindowId())
+        return window.getFocusId()
+
+    def setViewId(self, viewId):
```
Comment (enen92) [inline plugin.video.mediathekview/resources/lib/kodi/kodiaddon.py:113]: This is also not allowed

File: plugin.video.mediathekview/resources/lib/updateFileDownload.py (hunk 2 above)
Comment (codingPF) [reply plugin.video.mediathekview/resources/lib/updateFileDownload.py:1]: @enen92 
@Lunatixz
Thank you for your comments and review. I totally agree to your concerns and your efforts put into KODI to control the ecosystem is highly appreciated. Before going into some discussion on the technical details, I would like to clear out your concerns on the legitimate ownership. This plugin is part of the [mediathekview](https://mediathekview.de) organization. We do provide public, open and legal links to public Stations in one condensed API. This is explicitly limited to free media only. The Organization and this approach is well known in Germany and our Software and approach is regularly discussed in IT related papers e.g. [legal question](https://tipps.computerbild.de/software/systemsoftware/mediathekview-legal-ist-die-nutzung-erlaubt-621103.html), [Heise](https://www.heise.de/download/product/mediathekview-60659), [Computerbild](https://www.computerbild.de/download/MediathekView-4503190.html), [Pcwelt](https://www.pcwelt.de/downloads/Web-TV-MediathekView-583413.html), [Giga](https://www.giga.de/downloads/mediathekview/), [Wiki](https://de.wikipedia.org/wiki/MediathekView)
Mediathekview highly respects the ownership of intellectual property and does not support piracy or any kind of copyright violations.
Please let me know if this clarified the situation and if it makes sense to go into a technical discussion to align with KODI guidelines.

**Comment by Lunatixz** [comment]
@codingPF In order to move forward with merging you'll have to address a few issues with your code. Unfortunately, they were overlooked during the initial review; and allowed into the official repository. This is why we have temporarily marked your past Leia/matrix versions as broken. We want to work with you to resolve these issues in a manner that works best for both parties.

To comply with repository rules you'll need to remove the self-updating database. You have one of two options: include all parsers within the code and cache the database locally or provide a static cache file that must be included with the submission and only updated via repository updates. Your cache/database must be non-binary (csv,XML,json,etc). However, if you chose to include all parsers you can build a binary DB. locally.

The key issue is self-updating content through a third-party. This circumvents the submission process and is not allowed in most cases. Exceptions made to legitimate copyright owners and their backends. ie (API's, etc).

**Comment by codingPF** [comment]
@Lunatixz Thank you for taking this one up and pointing out options.  Just to make sure we are all on the same page.  The plugin provides open and legal content from several stations. The number of stations is known and limited to official, legal, public entities. Technically, the data itself is concentrated through some “proxy” (our backend) and brought to the plugin in some kind of mass update - to save performance for small devices (e.g. RP1,RP2). To have the update processes in a more elegant way (API,REST,..) is possible but this would still point to our proxy to do the job. So, the question would be if this is a valid exception from your point of view?

**Comment by Lunatixz** [comment]
>...is possible but this would still point to our proxy to do the job. So, the question would be if this is a valid exception from your point of view?

Unfortunately no, it is no allowed. The heart of the issue is proxy. Your API middle men's the content providers backend and circumvents the Kodi submission process by providing content that can not be easily verified.

The aforementioned solutions are your only options...

Properly implemented there is no reason why performance is a issue. Even for low power devices.

**Comment by alex1702** [comment]
Hi, I am one of the administrators of MediathekView. To describe it in more detail, we don't play proxy for the videos at any time. We are like a search engine, which collects the links to the shows and provides them in a list.

So if I understand your reasoning correctly, you would also reject a Google AddOn directly from Alphabet, since they also provide a list of publicly accessible links.

**Comment by Lunatixz** [comment]
> Hi, I am one of the administrators of MediathekView. To describe it in more detail, we don't play proxy for the videos at any time. We are like a search engine, which collects the links to the shows and provides them in a list.
> 
> So if I understand your reasoning correctly, you would also reject a Google AddOn directly from Alphabet, since they also provide a list of publicly accessible links.

Google does not distribute videos, it directs you to a site where you can view them. You are acting as a proxy to the content providers, Bypassing their site completely....

If you want to act as middlemen to others' content; you're required to show all work and include parsers from the source.

Every plugin allowed in this repository behaves in this manner or provides a static non-binary from which it can build from. The exceptions being API's from companies that OWN the rights to the media.

**Comment by codingPF** [comment]
> If you want to act as middlemen to others' content; you're required to show all work and include parsers from the source.

This is actually an interesting point. I am not sure if this point was mentioned properly. All this is open source. All sources are open and hosted on GITHUB, Everybody can build or review the source.
Maybe we can include a link to the Repo for review within the submission. Would that help?

**Comment by Lunatixz** [comment]
> > If you want to act as middlemen to others' content; you're required to show all work and include parsers from the source.
> 
> This is actually an interesting point. I am not sure if this point was mentioned properly. All this is open source. All sources are open and hosted on GITHUB, Everybody can build or review the source.
> Maybe we can include a link to the Repo for review within the submission. Would that help?

Team Kodi is only interested in the code that is submitted to the repository for review. It is not in our perview to evaluate your backend code.

You did not submit your parsing code with this project, only it's need to download a single database file. This file is the sole issue, it's contents cannot be verified because of its dynamic nature. It is not repository compliant.

Either correct the issue with one of the given solutions or close the request.

Thanks

**Comment by codingPF** [comment]
Hi Lunatixz, thank you for your comments. I think we all made our points clear. We will need to check what’s possible effort wise  and technically (py3 vs java etc). This will take some time…
Thanks for your time and efforts put into the discussion.

**Pull request closed without merging**